### PR TITLE
chore: improve changelog isVersionBump & commitID replacement regex

### PR DESCRIPTION
**Pull request by alvaromartmart**
Minor tweaks to the regex used to detect commit id & version bump commits

#### Checklist

- [x] Use a separate branch in your local repo (not `main`).
- [x] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

File: scripts/changelog.js [hunk 1]
```diff
@@ -51,14 +51,15 @@ const formatChanges = (changes) =>
         .replace(/[a-z]+\((\w+)\)/, '$1')
         // Linkify commit refs.
         .replace(
-          /^(\w{8})/,
+          /^(\w+)\s/,
```
Comment (delucis) [inline scripts/changelog.js:54]: I don’t think we want to replace the whitespace here.
```suggestion
          /^(\w+)/,
```

**Review by delucis — CHANGES_REQUESTED** [review]

File: scripts/changelog.js (hunk 1 above)
Comment (alvaromartmart) [reply scripts/changelog.js]: You're right, in fact I was re-introducing that space a couple of lines later. Fixed it, thanks!

File: scripts/changelog.js [hunk 2]
```diff
@@ -51,14 +51,15 @@ const formatChanges = (changes) =>
...
-          /^(\w{8})/,
+          /^(\w+)/,
           '* [[$1](https://github.com/boardgameio/boardgame.io/commit/$1)]'
         )
         // Linkify PR references.
         .replace(
           /\(#(\d{3,})\)/,
           '([#$1](https://github.com/boardgameio/boardgame.io/pull/$1))'
         )
+        .replace(/\)](\w*)/, ')]$1')
```
Comment (delucis) [inline scripts/changelog.js:62]: It's not always super easy to understand what a regex is intended to do. What's the intention of this addition?

Comment (alvaromartmart) [reply scripts/changelog.js:62]: Oops, you're right 😅  that line is no longer necessary as we no longer need to re-introduce the space between `[commidID]` and `commit message`

Comment (delucis) [reply scripts/changelog.js]: Ahhh, ok! Gotcha 😄

**Review by delucis — APPROVED** [review]

**Pull request merged**